app/controllers/concerns/current_project.rb

```diff
@@ -15,6 +15,7 @@ def current_project
 
   def require_project
     @project = (Project.find_by_param!(params[:project_id]) if params[:project_id])
+    @project ||= (Project.find_by_param!(params[:id]) if params[:id])
```

Comment:
This might cause problems on the stages controller, as it uses this function as well. I.e. If user tries to access a project page with a bad project_id param, it'll fall back to trying to find a project with the stages' param[:id].

I don't like it, but the solution is probably to do what the LocksController does and override the require_project.